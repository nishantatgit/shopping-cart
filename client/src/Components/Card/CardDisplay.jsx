import React from 'react';
import {Link} from 'react-router-dom';

export const CardDisplay = props => {
    return(
        <div className="card category-card text-center col-sm-4 position-relative">
            <div className="card-body d-flex flex-column">
                <div className="d-flex justify-content-center align-items-center mb-2">
                    <h5 className="card-title mb-0 mr-3">{props.title}</h5>
                    <img aria-hidden="true" src={`${window.location.origin}/svg/${props.pcatid}.svg`}/>
                </div>
                <p className="card-text">{props.desc}</p>
                <Link to={`products/${props.pcatid}`} className="btn bg-info text-white w-45 ml-auto mt-auto mr-auto">Browse</Link>
            </div>
        </div>
    );
}
