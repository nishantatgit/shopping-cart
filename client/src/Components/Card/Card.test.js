import React from 'react';
import {shallow} from 'enzyme';
import {CardDisplay} from './CardDisplay.jsx';

describe('CardDisplay', () => {
    it('should render correctly without props', () => {
        const component = shallow(<CardDisplay />);
        expect(component).toMatchSnapshot();
    });
    it('should render correctly with props', () => {
        const component = shallow(<CardDisplay title="product title" desc="this is some description" pcatid="pcat-1" />);
        expect(component).toMatchSnapshot();
    });
});