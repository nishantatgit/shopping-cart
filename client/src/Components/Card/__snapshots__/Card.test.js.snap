// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDisplay should render correctly with props 1`] = `
<div
  className="card category-card text-center col-sm-4 position-relative"
>
  <div
    className="card-body d-flex flex-column"
  >
    <div
      className="d-flex justify-content-center align-items-center mb-2"
    >
      <h5
        className="card-title mb-0 mr-3"
      >
        product title
      </h5>
      <img
        aria-hidden="true"
        src="http://localhost/svg/pcat-1.svg"
      />
    </div>
    <p
      className="card-text"
    >
      this is some description
    </p>
    <Link
      className="btn bg-info text-white w-45 ml-auto mt-auto mr-auto"
      replace={false}
      to="products/pcat-1"
    >
      Browse
    </Link>
  </div>
</div>
`;

exports[`CardDisplay should render correctly without props 1`] = `
<div
  className="card category-card text-center col-sm-4 position-relative"
>
  <div
    className="card-body d-flex flex-column"
  >
    <div
      className="d-flex justify-content-center align-items-center mb-2"
    >
      <h5
        className="card-title mb-0 mr-3"
      />
      <img
        aria-hidden="true"
        src="http://localhost/svg/undefined.svg"
      />
    </div>
    <p
      className="card-text"
    />
    <Link
      className="btn bg-info text-white w-45 ml-auto mt-auto mr-auto"
      replace={false}
      to="products/undefined"
    >
      Browse
    </Link>
  </div>
</div>
`;
